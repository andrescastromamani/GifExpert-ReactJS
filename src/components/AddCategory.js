import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const handleAddCategory = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 0) {
            setCategories(cats => [inputValue, ...cats])
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                placeholder={inputValue}
                value={inputValue}
                onChange={handleAddCategory} />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}