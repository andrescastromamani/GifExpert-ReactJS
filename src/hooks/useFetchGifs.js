import { useState } from "react"


export const useFetchGifs = () => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });
    setTimeout(() => {
        setState({
            data: [
                {
                    id: "ABC",
                    title: "ABC"
                },
                {
                    id: "DEF",
                    title: "DEF"
                }
            ],
            loading: false
        })
    }, 3000)
    return state;
}
