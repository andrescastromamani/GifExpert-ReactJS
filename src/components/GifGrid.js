import React, { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
    /*const [images, setImages] = useState([])
    useEffect(() => {
        getGifs(category).then(setImages)
    }, [category])*/
    const { loading, data } = useFetchGifs();
    console.log(data);
    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    loading ? 'Loading....' : 'Listo'
                }
                {/*
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))*/
                }
            </div>
        </>
    )
}