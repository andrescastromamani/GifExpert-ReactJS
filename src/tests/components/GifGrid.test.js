import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('GifGrid', () => {
    const category = 'music';
    test('should render a function', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })
    test('should render', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: 'ABC123',
            url: 'https://localhost/cualquiercosa.jpg',
            title: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})