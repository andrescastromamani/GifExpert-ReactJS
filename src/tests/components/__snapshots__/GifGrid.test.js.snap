// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should render a function 1`] = `
<Fragment>
  <h3>
    music
  </h3>
  <p
    className="animate__backInDown"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
